dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1106,38 +1106,56 @@ private void provisionWorkflowVersionUploadFiles(ApiClient zendoClient, Deposit
         if (sourceFiles == null || sourceFiles.size() == 0) {
             LOG.warn("No source files found to zip when creating DOI");
             throw new CustomWebApplicationException("No source files found to"
                     + " upload when creating DOI. Zenodo requires at lease one file"
                     + " to be uploaded in order to create a DOI.", HttpStatus.SC_BAD_REQUEST);
         } else {
-            String fileName = workflow.getWorkflowPath().replaceAll("/", "-") + ".zip";
             OutputStream outputStream;
+            String versionOfWorkflow = workflowVersion.getName();
+            String filePrefix = workflow.getWorkflowPath().replaceAll("/", "-")
+                    + "_" + versionOfWorkflow + "_";
+            String fileSuffix = ".zip";
+            File zipFile;
+            try {
+                zipFile = File.createTempFile(filePrefix, fileSuffix);
+            } catch (IOException e) {
+                LOG.error("Could not create Zenodo temp upload zip file." + " Error is " + e.getMessage(), e);
+                throw new CustomWebApplicationException("Could not create Zenodo upload temp zip file"
+                        + " Error is " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+            }
+            // Delete the file at least when the program stops; we will try to
+            // delete it right after upload in a deposit to Zenodo anyway
+            zipFile.deleteOnExit();
+
             try {
-                outputStream = new FileOutputStream(fileName);
+                outputStream = new FileOutputStream(zipFile);
             } catch (FileNotFoundException fne) {
-                LOG.error("Could not create file " + fileName
+                zipFile.delete();
+                LOG.error("Could not create file " + zipFile.getAbsolutePath()
                         + " outputstream for DOI zip file for upload to Zenodo."
                         + " Error is " + fne.getMessage(), fne);
-                throw new CustomWebApplicationException("Could not create file " + fileName
-                        + " outputstream for DOI zip file for upload to Zenodo."
+                throw new CustomWebApplicationException("Could not create "
+                        + "outputstream for DOI zip file for upload to Zenodo."
                         + " Error is " + fne.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
             }
-            writeStreamAsZip(sourceFiles, outputStream, path);
-            java.nio.file.Path zipPath = Paths.get(fileName);
-            File zipFile = zipPath.toFile();
+
+            String zipFileDir = zipFile.getParent();
+            java.nio.file.Path zipFilePath = Paths.get(zipFileDir);
+            writeStreamAsZip(sourceFiles, outputStream, zipFilePath);
 
             try {
-                filesApi.createFile(depositionID, zipFile, fileName);
+                filesApi.createFile(depositionID, zipFile, zipFile.getName());
```

Comment:
Is this the name that Zenodo uses? You could remove the extraneous characters here.

Reply:
I tried this but the Zenodo API seems to ignore the last parameter...it seems to use the name provided in the File object in the second parameter

Reply:
I give it the proper file name now.